src/index.ts

```diff
@@ -136,17 +137,17 @@ type ForwardPaginationArguments = { first: number; after?: string }
 type BackwardPaginationArguments = { last: number; before?: string }
 type NoPaginationArguments = Record<string, unknown>
 
-function mergeDefaultOptions<Model, Cursor, CustomEdge extends Edge<Model>>(
-  pOptions?: Options<Model, Cursor, CustomEdge>
-): Required<Options<Model, Cursor, CustomEdge>> {
+type MergedOptions<Model, Cursor, Node, CustomEdge extends Edge<Node>> = Omit<Required<Options<Model, Cursor, Node, CustomEdge>>, 'nodeToEdge'>
+
+function mergeDefaultOptions<Model, Cursor, Node, CustomEdge extends Edge<Node>>(
+  pOptions?: Options<Model, Cursor, Node, CustomEdge>
+): MergedOptions<Model, Cursor, Node, CustomEdge> {
   return {
     getCursor: (node: Model) =>
       ({ id: (node as unknown as { id: string }).id } as unknown as Cursor),
     encodeCursor: (cursor: Cursor) => (cursor as unknown as { id: string }).id,
     decodeCursor: (cursorString: string) => ({ id: cursorString } as unknown as Cursor),
-
-    nodeToEdge: (node) => ({ node } as Omit<CustomEdge, 'cursor'>),
-
+    modelToEdge: pOptions?.nodeToEdge ?? ((node: Model) => ({ node } as unknown as Omit<CustomEdge, 'cursor'>)),
```

Comment:
- Technically the change with the generics [here](https://github.com/devoxa/prisma-relay-cursor-connection/pull/245/files?file-filters%5B%5D=.ts#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R14-R15) is already a breaking version. So let's just keep it clean, kick out `nodeToEdge` entirely, and I'll release it as `2.0.0`.
- Replace mentions of `node: Model` with `model: Model` (also in other spots), if we're cleaning up.

Reply:
Yeah, you are right I don't usually explicitly denote generic classes and rely on TS to infer them. But this is a breaking change since we will get errors if types are explicitly set. I will remove nodeToEdge and update every mention as record (based on your comment above) and update the PR.